bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ChannelItemProvider.java

```diff
@@ -182,12 +183,23 @@ protected synchronized void modified(Map<String, Object> properties) {
             boolean initialDelay = properties == null
                     || !"false".equalsIgnoreCase((String) properties.get("initialDelay"));
             if (initialDelay) {
-                delayedInitialize(Executors.newSingleThreadScheduledExecutor());
+                executor = Executors.newSingleThreadScheduledExecutor();
+
+                linkRegistry.addRegistryChangeListener(linkRegistryLastUpdateListener);
```

Comment:
The listeners that are added by `addRegistryChangeListeners()` have additional logic e.g. creating or removing items. This additional logic should not be executed until the component is initialized. We can use these listeners instead of registering and deregistering new ones but then we need to modify them like
```
if (initialized) {
// old logic
} else {
lastUpdate = System.nanoTime();
}
```
Which approach do you prefer?

Reply:
Instead of creating new listeners just for the purpose of setting `lastUpdate` I would actually prefer using the existing listeners and implement the `if` condition you just sketched.

Reply:
I changed the implementation to use the existing listeners. I also changed the assertions in the test to because the old assertions didn't make sense with the new implementation.